include/swift/SIL/SILRemarkStreamer.h

```diff
@@ -0,0 +1,111 @@
+//===--- SILRemarkStreamer.h - Interface for streaming remarks --*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2020 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+/// \file
+/// This file defines the interface used to stream SIL optimization diagnostics
+/// through LLVM's RemarkStreamer interface.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef SWIFT_SIL_SILREMARKSTREAMER_H
+#define SWIFT_SIL_SILREMARKSTREAMER_H
+
+#include "swift/Demangling/Demangler.h"
+#include "swift/SIL/OptimizationRemark.h"
+#include "llvm/Remarks/RemarkStreamer.h"
+
+namespace swift {
+
+class SILRemarkStreamer {
+  llvm::remarks::RemarkStreamer &RS;
+  // Source manager for resolving source locations.
+  const SourceManager &SM;
+  // Temporary objects to avoid allocating memory for every remark when building
+  // strings.
+  mutable std::string TmpRemarkName;
```

Comment:
I really dislike this mutable usage here. I don't understand why you need it. Can you do it in a different way. If you need a temporary to build strings, use a SmallString. That gives you 64 characters.

Reply:
This was needed to add the `sil.` prefix to the pass name and to have a buffer to demangle the function name before it got serialized (through a StringRef). I moved all these to the creation of the remark now, so we shouldn't need this anymore.